okhttp/src/main/java/okhttp3/internal/connection/RealConnection.java

```diff
@@ -276,6 +284,9 @@ private void connectTls(ConnectionSpecSelector connectionSpecSelector) throws IO
             + "\n    subjectAltNames: " + OkHostnameVerifier.allSubjectAltNames(cert));
       }
 
+      X509Certificate cert = (X509Certificate) unverifiedHandshake.peerCertificates().get(0);
+      subjectAlternativeNames = OkHostnameVerifier.allSubjectAltNames(cert);
```

Comment:
ideally we compute this once before line 278, then reuse the parsed subjectAlternativeNames in the call to `verify()`